test/Constraints/closures.swift

```diff
@@ -875,3 +875,28 @@ struct rdar43866352<Options> {
     callback = { (options: Options) in } // expected-error {{cannot assign value of type '(inout Options) -> ()' to type '(inout _) -> Void'}}
   }
 }
+
+extension Hashable {
+  var self_: Self {
```

Comment:
Side note: Did this test case come from a user bug report? We could tell them they can use `\.self` key paths now :)

Reply:
This bug is from October, I'll let people who created this bug report know that there is `\.self` now.